src/status_im/ui/components/list_item/views.cljs

```diff
@@ -1,82 +1,284 @@
+        second-last-accessory       (peek (pop accessories))]
+    (vec
+     (concat
+      [react/view {:align-self :stretch
+                   :flex-direction :row
+                   :align-items :center
+                   :justify-content :flex-start}]
       (for [accessory accessories]
         (when-not (nil? accessory)
           (with-meta
             (cond
               (= :chevron accessory)
-              [react/view {:margin-right 8}
-               [icons/icon :main-icons/next {:color colors/gray-transparent-40}]]
+              [react/view
+               [icons/icon
+                :main-icons/next
+                {:container-style {:width           9
+                                   :height          16
+                                   :justify-content :center}
+                 :width           10
+                 :resize-mode     :center
+                 :height          16
+                 :color           colors/gray-transparent-40}]]
+
               (= :check accessory)
-              [react/view {:margin-right 16}
+              [react/view {:background-color colors/gray-transparent-40}
                [icons/icon :main-icons/check {:color colors/gray}]]
+
               (= :more accessory)
-              [react/view {:margin-right 16}
+              [react/view
                [icons/icon :main-icons/more]]
+
               :else
-              [react/view {:margin-right 16 :flex-shrink 1}
+              [react/view {:margin-right
+                           (if (= accessory last-accessory)
+                             0
+                             (if (and last-accessory-is-component
+                                      (string? second-last-accessory))
+                               1
+                               16))}
                (cond
                  (string? accessory)
-                 [react/text {:style styles/accessory-text :number-of-lines 1}
+                 [react/text {:style           styles/accessory-text
+                              :number-of-lines 1}
                   accessory]
+
                  (vector? accessory)
                  accessory
+
                  :else
                  [accessory])])
-            {:key accessory})))
-      (when error
-        [tooltip/tooltip error styles/error])]]))
+            {:key accessory})))))))
+
+(defn list-item
+  [{:keys [type theme container-margin-top container-margin-bottom image image-path
+           icon title title-color-override title-prefix title-prefix-width
+           title-prefix-height title-row-accessory subtitle subtitle-max-lines
+           subtitle-row-accessory content accessories disabled?
+           on-press on-long-press error accessibility-label]
+    :or   {type :default theme :default disabled? false
+           container-margin-top 0 container-margin-bottom 0
+           subtitle-max-lines 0}}]
+  (let [icon?           (or image icon image-path)
```

Comment:
hm maybe we could use only  `image` prop. and check if it vector or string, because only one can be used, wdyt?

Reply:
indeed - I cleaned this up locally but went with `:icon` to encompass all three, is that(naming) ok?

I figure as long as I document it, should be ok

Reply:
done